roles/ovirt-cluster-upgrade/tasks/main.yml

```diff
@@ -0,0 +1,128 @@
+---
+- block:
+  - name: Login to oVirt
+    ovirt_auth:
+      url: "{{ engine_url }}"
+      username: "{{ engine_user }}"
+      password: "{{ engine_password }}"
+      ca_file: "{{ engine_cafile | default(omit) }}"
+      insecure: "{{ engine_insecure | default(true) }}"
+    when: ovirt_auth is undefined
+    register: loggedin
+    tags:
+      - always
+
+  - name: Get hosts
+    ovirt_hosts_facts:
+      auth: "{{ ovirt_auth }}"
+      pattern: "cluster={{ cluster_name | mandatory }} and update_available=true"
+    tags:
+      - ovirt-cluster-upgrade
+  
+  - name: Check if there are hosts to be updated
+    debug:
+      msg: "No hosts to be updated"
+    when: ovirt_hosts | length == 0
+  
+  - block:
+    - name: Get cluster facts
+      ovirt_clusters_facts:
+        auth: "{{ ovirt_auth }}"
+        pattern: "name={{ cluster_name }}"
+      tags:
+        - ovirt-cluster-upgrade
```

Comment:
why it is needed to tag all of the tasks with 'ovirt-cluster-upgrade' if they are already part of the 'ovirt-cluster-upgrade' role ?